### ISSUE TITLE
foreach doesn't properly format the query string

### ISSUE THREAD

**Issue by dan-kez** [Yomguithereal/decypher#32]
``` js
var cypher = new Query();
cypher.foreach('name in {names}', {names: ['a']}).merge('(c {name: name})');
cypher.toString(); // 'FOREACH name in {names}\nMERGE (c:colPHI {name: name});'
```

Foreach should have the following format:

FOREACH ( `iterator` in `list` | `list of statements`)

**Comment by Yomguithereal** [comment]
That's an interesting issue @dmk255. I should look into it and adapt the API to this issue I think.

**Comment by dan-kez** [comment]
Thanks for the quick response! It'd be great if the API could support this.

**Comment by Yomguithereal** [comment]
What I propose to you is to add an argument to the forEach method taking the inner statement of the loop. Something along this:

``` js
cypher.foreach('name in {names}', 'MERGE (c {name: name})', {names: ['a']});
```

What do you think?

**Comment by dan-kez** [comment]
That definitely could work. I think having the option to put another query
object in would also be great.

E.g
subquery.merge('MERGE (c {name: name})');

cypher.foreach('name in {names}', subquery, {names: ['a']});

This would also combine the parameter objects from the subquery and the
parent query.

What do you think?

**Comment by Yomguithereal** [comment]
The only thing I need to know here, and excuse my rusty cypher, is whether the inner statement you can pass to the `FOREACH` can be more complex than a single statement?

**Comment by dan-kez** [comment]
Yes it can. There are some limitations on what you can do in the inner
query. For example you can't use the CALL. I'm not sure of all the
limitations though.

**Comment by Yomguithereal** [comment]
Ok. Do you have other statements in mind that behave like the `FOREACH` one? So I can handle them too.

**Comment by dan-kez** [comment]
Ther are a few that are similar but not the same. Take a look at this
document:

http://neo4j.com/docs/pdf/neo4j-cypher-refcard-stable.pdf

**Comment by Yomguithereal** [comment]
Just published v0.9.0 on npm @dmk255. Tell me whether this fits your needs and don't hesitate to come back if you find some more things needing to get tweaked.
